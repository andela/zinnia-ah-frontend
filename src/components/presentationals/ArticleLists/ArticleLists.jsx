import React from 'react';
import PropTypes from 'prop-types';
import { Item, Icon, Modal, Dimmer, Loader } from 'semantic-ui-react';
import moment from 'moment';

// config
import { DEFAULT_ARTICLE_IMAGE_URL } from '../../../utils/config';

// components
import Button from '../Button/Button';
import Title from '../Title/Title';

const ArticleLists = ({ articles, deleteArticle, isDeleting }) => {
  return (
    <div className="post-lists">
      {articles.length === 0 ? (
        <p>You have no publications</p>
      ) : (
        articles.map(article => (
          <Item.Group key={article.id}>
            <Item>
              <Item.Image
                src={article.imageThumbnail || DEFAULT_ARTICLE_IMAGE_URL}
              />

              <Item.Content>
                <div className="d-flex">
                  <div style={{ marginRight: 'auto' }}>
                    <Title
                      content={article.title}
                      className="title-article-lg"
                    />
                  </div>
                  <div style={{ marginRight: '0.8rem' }}>
                    <p
                      style={{
                        fontSize: '1rem',
                        width: '100%',
                      }}
                    >
                      {article.readTime} mins read
                    </p>
                  </div>
                  <div>
                    <Button className="btn-transparent" type="button" value="">
                      <Icon
                        name="bookmark outline"
                        style={{
                          fontSize: '1.25rem',
                          margin: '0',
                          color: '#000000',
                        }}
                      />
                    </Button>
                  </div>
                </div>
                <Item.Description
                  style={{
                    marginBottom: 'auto',
                  }}
                >
                  <p
                    style={{
                      fontSize: '1.125rem',
                      lineHeight: '1.375rem',
                      fontWeight: '200',
                    }}
                  >
                    {article.description}
                  </p>
                </Item.Description>
                <Item.Extra>
                  <div className="d-flex justify-content-between">
                    <p style={{ fontSize: '1rem' }}>
                      {moment().from(article.createdAt, 'YYYYMMDD')} ago
                    </p>
                    <div>
                      <Modal
                        trigger={
                          <Button
                            className="btn-transparent"
                            type="button"
                            value=""
                          >
                            <Icon
                              name="trash"
                              style={{
                                fontSize: '1.5rem',
                                margin: '0',
                                color: '#ff0000',
                              }}
                            />
                          </Button>
                        }
                        closeIcon
                        size="tiny"
                      >
                        {isDeleting && (
                          <Dimmer active>
                            <Loader />
                          </Dimmer>
                        )}
                        <Modal.Header>Delete Article!</Modal.Header>
                        <Modal.Content>
                          <p>Are you sure you want to delete this Article?</p>
                        </Modal.Content>
                        <Modal.Actions>
                          <Button
                            className="btn-dark mb-3"
                            type="button"
                            value="DELETE"
                            key="delete"
                            onClick={() => deleteArticle(article.id, articles)}
                          />
                        </Modal.Actions>
                      </Modal>
                    </div>
                  </div>
                </Item.Extra>
              </Item.Content>
            </Item>
          </Item.Group>
        ))
      )}
    </div>
  );
};

ArticleLists.propTypes = {
  articles: PropTypes.array,
  deleteArticle: PropTypes.any,
  isDeleting: PropTypes.bool,
};

export default ArticleLists;
