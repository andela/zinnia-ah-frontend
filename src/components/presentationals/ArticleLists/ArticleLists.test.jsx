import React from 'react';
import { mount } from 'enzyme';

import ArticleLists from './ArticleLists';

const props = {
  deleteArticle: jest.fn(),
  isDeleting: false,
  articles: [{}, {}],
};

describe('ArticleLists', () => {
  it('should render correctly', () => {
    const component = mount(<ArticleLists {...props} />);
    expect(component).toMatchSnapshot();
  });
});
