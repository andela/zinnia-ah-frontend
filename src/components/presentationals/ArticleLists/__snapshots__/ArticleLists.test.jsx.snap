// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleLists should render correctly 1`] = `
<ArticleLists
  articles={
    Array [
      Object {},
      Object {},
    ]
  }
  deleteArticle={[Function]}
  isDeleting={false}
>
  <div
    className="post-lists"
  >
    <ItemGroup>
      <div
        className="ui items"
      >
        <Item>
          <div
            className="item"
          >
            <ItemImage
              src="https://react.semantic-ui.com/images/wireframe/image.png"
            >
              <Image
                as="img"
                src="https://react.semantic-ui.com/images/wireframe/image.png"
                ui={false}
                wrapped={true}
              >
                <div
                  className="image"
                >
                  <img
                    src="https://react.semantic-ui.com/images/wireframe/image.png"
                  />
                </div>
              </Image>
            </ItemImage>
            <ItemContent>
              <div
                className="content"
              >
                <div
                  className="d-flex"
                >
                  <div
                    style={
                      Object {
                        "marginRight": "auto",
                      }
                    }
                  >
                    <Title
                      className="title-article-lg"
                    >
                      <h1
                        className="title title-article-lg"
                      />
                    </Title>
                  </div>
                  <div
                    style={
                      Object {
                        "marginRight": "0.8rem",
                      }
                    }
                  >
                    <p
                      style={
                        Object {
                          "fontSize": "1rem",
                          "width": "100%",
                        }
                      }
                    >
                      2 mins read
                    </p>
                  </div>
                  <div>
                    <Button
                      className="btn-transparent"
                      type="button"
                      value=""
                    >
                      <button
                        className="btn btn-transparent"
                        type="button"
                      >
                        <Icon
                          as="i"
                          name="bookmark outline"
                          style={
                            Object {
                              "color": "#000000",
                              "fontSize": "1.25rem",
                              "margin": "0",
                            }
                          }
                        >
                          <i
                            aria-hidden="true"
                            className="bookmark outline icon"
                            onClick={[Function]}
                            style={
                              Object {
                                "color": "#000000",
                                "fontSize": "1.25rem",
                                "margin": "0",
                              }
                            }
                          />
                        </Icon>
                      </button>
                    </Button>
                  </div>
                </div>
                <ItemDescription
                  style={
                    Object {
                      "marginBottom": "auto",
                    }
                  }
                >
                  <div
                    className="description"
                    style={
                      Object {
                        "marginBottom": "auto",
                      }
                    }
                  >
                    <p
                      style={
                        Object {
                          "fontSize": "1.125rem",
                          "fontWeight": "200",
                          "lineHeight": "1.375rem",
                        }
                      }
                    />
                  </div>
                </ItemDescription>
                <ItemExtra>
                  <div
                    className="extra"
                  >
                    <div
                      className="d-flex justify-content-between"
                    >
                      <p
                        style={
                          Object {
                            "fontSize": "1rem",
                          }
                        }
                      >
                        3 days ago
                      </p>
                      <div>
                        <Modal
                          centered={true}
                          closeIcon={true}
                          closeOnDimmerClick={true}
                          closeOnDocumentClick={false}
                          dimmer={true}
                          eventPool="Modal"
                          size="tiny"
                          trigger={
                            <Button
                              className="btn-transparent"
                              type="button"
                              value=""
                            >
                              <Icon
                                as="i"
                                name="trash"
                                style={
                                  Object {
                                    "color": "#ff0000",
                                    "fontSize": "1.5rem",
                                    "margin": "0",
                                  }
                                }
                              />
                            </Button>
                          }
                        >
                          <Portal
                            closeOnDocumentClick={false}
                            closeOnEscape={true}
                            eventPool="Modal"
                            mountNode={<body />}
                            onClose={[Function]}
                            onMount={[Function]}
                            onOpen={[Function]}
                            onUnmount={[Function]}
                            openOnTriggerClick={true}
                            trigger={
                              <Button
                                className="btn-transparent"
                                type="button"
                                value=""
                              >
                                <Icon
                                  as="i"
                                  name="trash"
                                  style={
                                    Object {
                                      "color": "#ff0000",
                                      "fontSize": "1.5rem",
                                      "margin": "0",
                                    }
                                  }
                                />
                              </Button>
                            }
                          >
                            <Ref
                              innerRef={[Function]}
                            >
                              <RefFindNode
                                innerRef={[Function]}
                              >
                                <Button
                                  className="btn-transparent"
                                  onBlur={[Function]}
                                  onClick={[Function]}
                                  onFocus={[Function]}
                                  onMouseEnter={[Function]}
                                  onMouseLeave={[Function]}
                                  type="button"
                                  value=""
                                >
                                  <button
                                    className="btn btn-transparent"
                                    onClick={[Function]}
                                    type="button"
                                  >
                                    <Icon
                                      as="i"
                                      name="trash"
                                      style={
                                        Object {
                                          "color": "#ff0000",
                                          "fontSize": "1.5rem",
                                          "margin": "0",
                                        }
                                      }
                                    >
                                      <i
                                        aria-hidden="true"
                                        className="trash icon"
                                        onClick={[Function]}
                                        style={
                                          Object {
                                            "color": "#ff0000",
                                            "fontSize": "1.5rem",
                                            "margin": "0",
                                          }
                                        }
                                      />
                                    </Icon>
                                  </button>
                                </Button>
                              </RefFindNode>
                            </Ref>
                          </Portal>
                        </Modal>
                      </div>
                    </div>
                  </div>
                </ItemExtra>
              </div>
            </ItemContent>
          </div>
        </Item>
      </div>
    </ItemGroup>
    <ItemGroup>
      <div
        className="ui items"
      >
        <Item>
          <div
            className="item"
          >
            <ItemImage
              src="https://react.semantic-ui.com/images/wireframe/image.png"
            >
              <Image
                as="img"
                src="https://react.semantic-ui.com/images/wireframe/image.png"
                ui={false}
                wrapped={true}
              >
                <div
                  className="image"
                >
                  <img
                    src="https://react.semantic-ui.com/images/wireframe/image.png"
                  />
                </div>
              </Image>
            </ItemImage>
            <ItemContent>
              <div
                className="content"
              >
                <div
                  className="d-flex"
                >
                  <div
                    style={
                      Object {
                        "marginRight": "auto",
                      }
                    }
                  >
                    <Title
                      className="title-article-lg"
                    >
                      <h1
                        className="title title-article-lg"
                      />
                    </Title>
                  </div>
                  <div
                    style={
                      Object {
                        "marginRight": "0.8rem",
                      }
                    }
                  >
                    <p
                      style={
                        Object {
                          "fontSize": "1rem",
                          "width": "100%",
                        }
                      }
                    >
                      2 mins read
                    </p>
                  </div>
                  <div>
                    <Button
                      className="btn-transparent"
                      type="button"
                      value=""
                    >
                      <button
                        className="btn btn-transparent"
                        type="button"
                      >
                        <Icon
                          as="i"
                          name="bookmark outline"
                          style={
                            Object {
                              "color": "#000000",
                              "fontSize": "1.25rem",
                              "margin": "0",
                            }
                          }
                        >
                          <i
                            aria-hidden="true"
                            className="bookmark outline icon"
                            onClick={[Function]}
                            style={
                              Object {
                                "color": "#000000",
                                "fontSize": "1.25rem",
                                "margin": "0",
                              }
                            }
                          />
                        </Icon>
                      </button>
                    </Button>
                  </div>
                </div>
                <ItemDescription
                  style={
                    Object {
                      "marginBottom": "auto",
                    }
                  }
                >
                  <div
                    className="description"
                    style={
                      Object {
                        "marginBottom": "auto",
                      }
                    }
                  >
                    <p
                      style={
                        Object {
                          "fontSize": "1.125rem",
                          "fontWeight": "200",
                          "lineHeight": "1.375rem",
                        }
                      }
                    />
                  </div>
                </ItemDescription>
                <ItemExtra>
                  <div
                    className="extra"
                  >
                    <div
                      className="d-flex justify-content-between"
                    >
                      <p
                        style={
                          Object {
                            "fontSize": "1rem",
                          }
                        }
                      >
                        3 days ago
                      </p>
                      <div>
                        <Modal
                          centered={true}
                          closeIcon={true}
                          closeOnDimmerClick={true}
                          closeOnDocumentClick={false}
                          dimmer={true}
                          eventPool="Modal"
                          size="tiny"
                          trigger={
                            <Button
                              className="btn-transparent"
                              type="button"
                              value=""
                            >
                              <Icon
                                as="i"
                                name="trash"
                                style={
                                  Object {
                                    "color": "#ff0000",
                                    "fontSize": "1.5rem",
                                    "margin": "0",
                                  }
                                }
                              />
                            </Button>
                          }
                        >
                          <Portal
                            closeOnDocumentClick={false}
                            closeOnEscape={true}
                            eventPool="Modal"
                            mountNode={<body />}
                            onClose={[Function]}
                            onMount={[Function]}
                            onOpen={[Function]}
                            onUnmount={[Function]}
                            openOnTriggerClick={true}
                            trigger={
                              <Button
                                className="btn-transparent"
                                type="button"
                                value=""
                              >
                                <Icon
                                  as="i"
                                  name="trash"
                                  style={
                                    Object {
                                      "color": "#ff0000",
                                      "fontSize": "1.5rem",
                                      "margin": "0",
                                    }
                                  }
                                />
                              </Button>
                            }
                          >
                            <Ref
                              innerRef={[Function]}
                            >
                              <RefFindNode
                                innerRef={[Function]}
                              >
                                <Button
                                  className="btn-transparent"
                                  onBlur={[Function]}
                                  onClick={[Function]}
                                  onFocus={[Function]}
                                  onMouseEnter={[Function]}
                                  onMouseLeave={[Function]}
                                  type="button"
                                  value=""
                                >
                                  <button
                                    className="btn btn-transparent"
                                    onClick={[Function]}
                                    type="button"
                                  >
                                    <Icon
                                      as="i"
                                      name="trash"
                                      style={
                                        Object {
                                          "color": "#ff0000",
                                          "fontSize": "1.5rem",
                                          "margin": "0",
                                        }
                                      }
                                    >
                                      <i
                                        aria-hidden="true"
                                        className="trash icon"
                                        onClick={[Function]}
                                        style={
                                          Object {
                                            "color": "#ff0000",
                                            "fontSize": "1.5rem",
                                            "margin": "0",
                                          }
                                        }
                                      />
                                    </Icon>
                                  </button>
                                </Button>
                              </RefFindNode>
                            </Ref>
                          </Portal>
                        </Modal>
                      </div>
                    </div>
                  </div>
                </ItemExtra>
              </div>
            </ItemContent>
          </div>
        </Item>
      </div>
    </ItemGroup>
  </div>
</ArticleLists>
`;
